observability/opentelemetry/v2/client/otel_observability_service.go

```diff
@@ -0,0 +1,197 @@
+		trace.WithSpanKind(trace.SpanKindConsumer),
+		trace.WithAttributes(attribute.String(string(semconv.CodeFunctionKey), "RecordReceivedMalformedEvent")))
+
+	recordSpanError(span, err)
+	span.End()
+}
+
+// RecordCallingInvoker starts a new span before calling the invoker upon a received event.
+// In case the operation fails, the error is recorded and the span is marked as failed.
+func (os OTelObservabilityService) RecordCallingInvoker(ctx context.Context, event *cloudevents.Event) (context.Context, func(errOrResult error)) {
+	spanName := os.getSpanName(event, "receive")
+	ctx, span := os.tracer.Start(
+		ctx, spanName,
+		trace.WithSpanKind(trace.SpanKindConsumer),
+		trace.WithAttributes(GetDefaultSpanAttributes(event, "RecordCallingInvoker")...))
+
+	if span.IsRecording() && os.spanAttributesGetter != nil {
+		span.SetAttributes(os.spanAttributesGetter(*event)...)
+	}
+
+	return ctx, func(errOrResult error) {
+		recordSpanError(span, errOrResult)
+		span.End()
+	}
+}
+
+// RecordSendingEvent starts a new span before sending the event.
+// In case the operation fails, the error is recorded and the span is marked as failed.
+func (os OTelObservabilityService) RecordSendingEvent(ctx context.Context, event cloudevents.Event) (context.Context, func(errOrResult error)) {
+	spanName := os.getSpanName(&event, "send")
+
+	ctx, span := os.tracer.Start(
+		ctx, spanName,
+		trace.WithSpanKind(trace.SpanKindProducer),
+		trace.WithAttributes(GetDefaultSpanAttributes(&event, "RecordSendingEvent")...))
+
+	if span.IsRecording() && os.spanAttributesGetter != nil {
+		span.SetAttributes(os.spanAttributesGetter(event)...)
+	}
+
+	return ctx, func(errOrResult error) {
+		recordSpanError(span, errOrResult)
+		span.End()
+	}
+}
+
+// RecordRequestEvent starts a new span before transmitting the given request.
+// In case the operation fails, the error is recorded and the span is marked as failed.
+func (os OTelObservabilityService) RecordRequestEvent(ctx context.Context, event cloudevents.Event) (context.Context, func(errOrResult error, event *cloudevents.Event)) {
+	spanName := os.getSpanName(&event, "send")
+
+	ctx, span := os.tracer.Start(
+		ctx, spanName,
+		trace.WithSpanKind(trace.SpanKindProducer),
+		trace.WithAttributes(GetDefaultSpanAttributes(&event, "RecordRequestEvent")...))
+
+	if span.IsRecording() && os.spanAttributesGetter != nil {
+		span.SetAttributes(os.spanAttributesGetter(event)...)
+	}
+
+	return ctx, func(errOrResult error, event *cloudevents.Event) {
+		recordSpanError(span, errOrResult)
+		span.End()
+	}
+}
+
+func GetDefaultSpanAttributes(e *cloudevents.Event, method string) []attribute.KeyValue {
+	attr := []attribute.KeyValue{
+		attribute.String(string(semconv.CodeFunctionKey), method),
+		attribute.String(observability.SpecversionAttr, e.SpecVersion()),
+		attribute.String(observability.IdAttr, e.ID()),
+		attribute.String(observability.TypeAttr, e.Type()),
+		attribute.String(observability.SourceAttr, e.Source()),
```

Comment:
What is the process for that? Just open a PR or should it be discussed first on a issue?

Reply:
If this was brand new it should probably be discussed first on an issue but since CloudEvents themselves already have the attributes defined on their end I think you can just create a PR towards otel-spec and reference `keys.go` from here (or the CloudEvents spec if it's there as well) as source.